lib/patch-series.ts

```diff
@@ -231,31 +262,69 @@ export class PatchSeries {
         return mbox.split(separatorRegex);
     }
 
-    protected static removeDuplicateMimeVersionHeaders(mails: string[]): void {
+    protected static removeDuplicateHeaders(mails: string[]): void {
         mails.map((mail: string, i: number) => {
             const endOfHeader = mail.indexOf("\n\n");
             if (endOfHeader < 0) {
                 return;
             }
+
             let headers = mail.substr(0, endOfHeader + 1);
-            const needle = "\nMIME-Version: 1.0\n";
-            const offset = headers.indexOf(needle);
-            if (offset < 0) {
-                return;
-            }
-            let offset2 = headers.indexOf(needle, offset + 1);
-            if (offset2 < 0) {
-                return;
-            }
-            do {
-                headers = headers.substr(0, offset2 + 1)
-                    + headers.substr(offset2 + needle.length);
-                offset2 = headers.indexOf(needle, offset + 1);
-            } while (offset2 > 0);
+            singletonHeaders.forEach((header: ISingletonHeader) => {
+                headers = PatchSeries.stripDuplicateHeaders(headers, header);
+            });
+
             mails[i] = headers + mail.substr(endOfHeader + 1);
         });
     }
 
+    protected static stripDuplicateHeaders(headers: string,
+                                           header: ISingletonHeader): string {
+        const needle = "\n" + header.key + ":";
+        let offset;
+
+        if (headers.startsWith(`${header.key}:`)) {
+            offset = 0;
+        } else {
+            offset = headers.indexOf(needle) + 1;
+            if (!offset) {
+                return headers;
+            }
+        }
+
+        let endofKey = offset + needle.length - 1;
+        offset = headers.indexOf(needle, endofKey + 1);
+
+        if (offset < 0) {
+            return headers;
+        }
+
+        // extract values to determine if they match.
+        let endOfHdr = headers.indexOf("\n", endofKey);
+        const value1 = headers.substr(endofKey, endOfHdr - endofKey).trim();
+
+        do {
+            endofKey = offset + needle.length;
```

Comment:
At this point, `offset` is pointing at the newline preceding the next instance of `key`, so `endofKey` points just after the trailing `:`. Okay.